include/mesh.h

```diff
@@ -0,0 +1,273 @@
+#ifndef PIPE_NETWORK_MESH_H_
+#define PIPE_NETWORK_MESH_H_
+
+#include <cmath>
+
+#include <array>
+#include <exception>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <vector>
+
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/betweenness_centrality.hpp>
+#include <boost/graph/graph_traits.hpp>
+
+#include "node.h"
+#include "pipe.h"
+
+//! Mesh class
+//! \brief Class for mesh that contains node and pipe pointers
+class Mesh {
+
+  typedef boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS>
+      graph;
+  typedef boost::graph_traits<graph>::vertex_descriptor vertex;
+  typedef boost::graph_traits<graph>::edge_descriptor edge;
+  typedef boost::property_map<graph, boost::vertex_index_t>::type vindexmap;
+  typedef std::map<edge, int> stdeindex;
+  typedef boost::associative_property_map<stdeindex> eindexmap;
+
+ public:
+  // Constructor with id
+  //! \param[in] id mesh id
+  Mesh(unsigned id) : id_{id} {};
+
+  //! Destructor
+  ~Mesh() {
+    nodes_.clear();
+    pipes_.clear();
+  };
+
+  //! Return id
+  //! \retval id_ id of the mesh
+  unsigned id() const { return id_; }
+
+  //! Create a nodal pointer based on input id and coordinates
+  //! \param[in] id the node id
+  //! \param[in] coords the coordinates of the node
+  void create_node(unsigned id, Eigen::Vector3d coords) {
+    nodes_.emplace_back(std::make_shared<pipenetwork::Node>(id, coords));
+  }
+
+  //! Check whether there are nodes with duplicate indices or coordinates
+  void duplicate_node() {
```

Comment:
Rewrite this using `range-based-for` loops and/or STL algorithms / Lambda functions